.github/workflows/beam_PreCommit_Go.yml

```diff
@@ -1,37 +1,57 @@
-name: beam_PreCommit_Go
+name: Go Precommit
 
 on:
   push:
     tags: ['v*']
     branches: ['master', 'release-*']
-  pull_request:
-    branches: ['master']
+  pull_request_target:
+    branches: ['master', 'release-*' ]
     paths: ['model/**', 'sdks/go.**', 'release/**']
   issue_comment:
     types: [created]
   schedule:
     - cron: '* */6 * * *'
+permissions:
```

Comment:
As we use `pull_request_target` event we also need to add `read-all` permissions based on the instruction: https://github.com/apache/beam/blob/master/.github/ACTIONS.md?plain=1#L35

Reply:
I cant put `permissions: read-all` because im using nested to add writes 
```
permissions: 
   actions: write
   pull-requests: read
 ```
 

We can do 
``` 
permissions:
  actions: write
  pull-requests: read
  checks: read
  contents: read
  deployments: read
  id-token: read
  issues: read
  discussions: read
  packages: read
  pages: read
  repository-projects: read
  security-events: read
  statuses: read
  ```
  
Which is a bit longer but it think is needed because it seems that for `pull_request_target` token is given write to all by default unless you specify otherwise with `permissions`

Reply:
Applied changes